src/xopen/__init__.py

```diff
@@ -441,6 +452,33 @@ def __init__(
         super().__init__(path, ["pigz"], mode, compresslevel, "-p", threads)
 
 
+class PipedPBzip2Reader(PipedCompressionReader):
+    """
+    Open a pipe to pbzip2 for reading a bzipped file.
+    """
+    def __init__(self, path, mode: str = "r", threads: Optional[int] = None):
+        super().__init__(path, ["pbzip2"], mode, "-p", threads, space=False)
+
+class PipedPBzip2Writer(PipedCompressionWriter):
+    """
+    Write bzip2-compressed files by running an external pbzip2 process and
+    piping into it. pbzip2 can compress using multiple cores.
+    """
+    _accepted_compression_levels: Set[int] = set(list(range(10)))
+
+    def __init__(
+        self,
+        path,
+        mode: str = "wt",
+        compresslevel: Optional[int] = None,
+        threads: Optional[int] = None,
+    ):
+        """
+        """
+        if compresslevel is not None and compresslevel not in self._accepted_compression_levels:
```

Comment:
can also be `compresslevel not in range(10)`. Much simpler. Pigz however accepts (0-9, 11) so it needed more complex code.

Reply:
Done!